_pages/vectorautodiff.md

```diff
@@ -0,0 +1,189 @@
+is explained in the below sections.
+
+## Working
+
+To demonstrate the working of standard forward mode AD, we have shown two versions 
+of the code block below. The initial snippet is the original function and later
+we have the generated function by clad for differentiating.
+
+```cpp
+double f(double x, double y, double z) {
+  return x + y + z;
+}
+
+int main() {
+  // Call clad to generate the derivative of f wrt x.
+  auto f_dx_dz = clad::differentiate(f, "x");
+
+  // Execute the generated derivative function.  
+  double dx = f_dx.execute(/*x=*/ 3, /*y=*/4, /*z=*/ 5);
+}
+```
+The `clad::differentiate` performs the derivative of the function `f` with respect 
+to `x`, it is used for automatic differentiation, and it generates a function that
+compute the derivative. It specifies `x` as the argument to indicate that it wants
+derivatives with respect to`x`.
+
+The above gets generated into below function generated by clad for differentiating:
+
+```cpp
+double f_darg0(double x, double y, double z) {
+  double _d_x = 1;
+  double _d_y = 0;
+  double _d_z = 0;
+  return _d_x + _d_y + _d_z;
+}
+```
+
+When we see this with vector mode, we can compute derivate w.r.t multiple params
+together shown below.
+
+Same function taken as the above example:
+
+```cpp
+double f(double x, double y, double z) {
+  return x + y + z;
+}
+
+int main() {
+  // Call clad to generate the derivative of f wrt x and z.
+  auto f_dx_dz = clad::differentiate <clad::opts::vector_mode> (f, "x,z");
+
+  // Execute the generated derivative function.  
+  double dx = 0, dy = 0, dz = 0;
+  f_dx.execute(/*x=*/ 3, /*y=*/4, /*z=*/ 5, &dx, &dz);
+}
+```
+The `clad::differentiate` performs the derivative of the function `f` with respect 
+to `x and z`, it is used for automatic differentiation, and it generates a
+function that computes the derivatives as shown below. It specifies `x,z` as the
+argument to indicate that it wants derivatives with respect to both `x and z`.
+The `<clad::opts::vector_mode>` has been used here.
+
+`f_dx.execute` calculates the derivatives and stores them in the dx and dz by
+giving their references.
+
+This above function gets generated into the below form:
+
+```cpp
+void f_dvec_0_2(double x, double y, double z, double *_d_x, double *_dz) {
+  clad::array<double> d_vec_x = {1, 0};
+  clad::array<double> d_vec_y = {0, 0}
+  clad::array<double> d_vec_z = {0, 1};
+  {
+    clad::array<double> d_vec_ret = d_vec_x + d_vec_y + d_vec_z;
+    *_d_x = d_vec_ret[0];
+    *_d_z = d_vec_ret[1];
+    return;
+  }
+}
+```
+
+**Note**: The main change in interface is using the `clad::opts::vector_mode`.
+
+## Defining the proposed solution
+
+For computing gradient of a function with -dimensional input (𝕟) - forward mode
+requires n forward passes.
+
+<div style="max-width:600px; margin:0 auto;">
+  <img src="/images/autodiff-forward.png" style="max-width:90%;"><br />
+<p align="center">
+Vectorized Automatic Differentiation
```

Comment:
The caption is wrong for this image. The image represents gradient computation using non-vectorized forward mode AD. A better caption would be "Non-Vectorized Forward Mode Automatic Differentiation".